File: cpp/226.cpp
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
  TreeNode* invertTree(TreeNode* root) {
    if(root == nullptr)
      return nullptr;
    auto t = new TreeNode(root->val);
    t->left = invertTree(root->right);
    t->right = invertTree(root->left);
    return t;
  }
};

File: python/226.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if root is None:
            return None
        t = TreeNode(root.val)
        t.left = self.invertTree(root.right)
        t.right = self.invertTree(root.left)
        return t
